### PR TITLE
Added Fly io deployment files

### PR DESCRIPTION
**Pull request by gowsik-ragunath**
What changed:

- Added docker and yaml files for deploying the app on fly.io
- Update preset-typescript to latest version

Why:

To deploy staging app in fly.io

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+COPY --from=gems /app /app
+COPY --from=node_modules /app/node_modules /app/node_modules
+COPY --from=gems /usr/lib/fullstaq-ruby/versions /usr/lib/fullstaq-ruby/versions
+COPY --from=gems /usr/local/bundle /usr/local/bundle
+
+# The following enable assets to precompile on the build server.  Adjust
+# as necessary.  If no combination works for you, see:
+# https://fly.io/docs/rails/getting-started/existing/#access-to-environment-variables-at-build-time
+ENV SECRET_KEY_BASE 1
+# ENV AWS_ACCESS_KEY_ID=1
```
Comment (vipulnsward) [inline Dockerfile:93]: How do we export these to App?

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "dependencies": {
     "@babel/plugin-proposal-private-methods": "^7.16.5",
     "@babel/preset-react": "^7.16.5",
-    "@babel/preset-typescript": "^7.16.5",
+    "@babel/preset-typescript": "7.21.0",
```
Comment (vipulnsward) [inline package.json:7]: This change is required for?

**Review by vipulnsward — APPROVED** [review]

**Comment by supriya3105** [comment]
@gowsik-ragunath Please respond to Vipul's query and merge this PR if its ready.

File: package.json (hunk 2 above)
Comment (gowsik-ragunath) [reply package.json:7]: No, we can revert this change. I updated the package while fixing the fly.io memory issue.

File: Dockerfile (hunk 1 above)
Comment (gowsik-ragunath) [reply Dockerfile:93]: This dummy key will be used only while building the docker image to build assets, but Fly will use the env keys we store as secrets to launch the app.

**Comment by gowsik-ragunath** [comment]
@supriya3105 we can merged this PR now address all the comments

**Pull request merged**